SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp

```diff
@@ -627,37 +634,136 @@ void ASpatialDebugger::DrawDebug(UCanvas* Canvas, APlayerController* /* Controll
-				{
-					continue;
-				}
+				DrawTag(Canvas, ScreenLocation, EntityId, Actor->GetName(), true /*bCentre*/);
+			}
+		}
+	}
+}
 
-				FVector2D ScreenLocation = FVector2D::ZeroVector;
-				if (LocalPlayerController.IsValid())
-				{
-					SCOPE_CYCLE_COUNTER(STAT_Projection);
-					UGameplayStatics::ProjectWorldToScreen(LocalPlayerController.Get(), ActorLocation + WorldSpaceActorTagOffset,
-														   ScreenLocation, false);
-				}
+void ASpatialDebugger::SelectActorToTag(UCanvas* Canvas)
+{
+	if (LocalPlayerController.IsValid())
+	{
+		FVector2D MousePosition;
 
-				if (ScreenLocation.IsZero())
+		if (LocalPlayerController->GetMousePosition(MousePosition.X, MousePosition.Y))
+		{
+			bool CursorChanged = false;
+			if (CrosshairTexture)
+			{
+				// Display a crosshair icon for the mouse cursor
+				// Offset by half of the texture's dimensions so that the center of the texture aligns with the center of the Canvas.
+				FVector2D CrossHairDrawPosition(MousePosition.X - (CrosshairTexture->GetSurfaceWidth() * 0.5f),
+												MousePosition.Y - (CrosshairTexture->GetSurfaceHeight() * 0.5f));
+
+				// Draw the crosshair at the mouse position.
+				FCanvasTileItem TileItem(CrossHairDrawPosition, CrosshairTexture->Resource, FLinearColor::White);
+				TileItem.BlendMode = SE_BLEND_Translucent;
+				Canvas->DrawItem(TileItem);
+				CursorChanged = true;
+			}
+
+			if (CursorChanged)
+			{
+				// Hide the mouse cursor as we will draw our own custom crosshair
+				LocalPlayerController->bShowMouseCursor = false;
+				// Sets back the focus to the game viewport - need to hide mouse cursor instantly
+				FSlateApplication::Get().SetAllUserFocusToGameViewport();
+			}
+
+			// Use the mouse position as the point for sending the raycast into the world
+			const APlayerCameraManager* CameraManager = Cast<APlayerCameraManager>(LocalPlayerController->PlayerCameraManager);
+			if (CameraManager)
+			{
+				FVector WorldLocation;
+				FVector WorldRotation;
+				LocalPlayerController->DeprojectScreenPositionToWorld(MousePosition.X, MousePosition.Y, WorldLocation,
+																	  WorldRotation); // Mouse cursor position
+				FVector StartTrace = WorldLocation;
+				FVector EndTrace = StartTrace + WorldRotation * MaxRange;
+
+				FCollisionObjectQueryParams CollisionObjectParams;
+				CollisionObjectParams.AddObjectTypesToQuery(ECollisionChannel::ECC_WorldStatic);
+				CollisionObjectParams.AddObjectTypesToQuery(ECollisionChannel::ECC_WorldDynamic);
+				CollisionObjectParams.AddObjectTypesToQuery(ECollisionChannel::ECC_Pawn);
+				CollisionObjectParams.AddObjectTypesToQuery(ECollisionChannel::ECC_Destructible);
+				CollisionObjectParams.AddObjectTypesToQuery(ECollisionChannel::ECC_Vehicle);
+				CollisionObjectParams.AddObjectTypesToQuery(ECollisionChannel::ECC_PhysicsBody);
+
+				TArray<FHitResult> HitResults;
+				bool hit = GetWorld()->LineTraceMultiByObjectType(HitResults, StartTrace, EndTrace, CollisionObjectParams);
```

Comment:
Might be good to add a comment in the header about how this will only work with actors that collision constraints setup

Reply:
Done